hamilton/caching/stores/sqlite.py

```diff
@@ -106,13 +107,21 @@ def set(
         self,
         *,
         cache_key: str,
-        node_name: str,
-        code_version: str,
         data_version: str,
         run_id: str,
+        node_name: str = None,
+        code_version: str = None,
```

Comment:
this could be backwards incompatible. But if we're the only ones using this, then all good.

Reply:
It's not because the `HamiltonCacheAdapter` is the only object passing values here. As shown in the updated logic of `.get()` it will resolve the values of `node_name` and `code_version` from the `cache_key` so these values will never be `None` (for instance, the SQL table still enforce `NOT NULL` constraints)

For instance, the `MetadataStore` base class doesn't expose `node_name` and `code_version` as arguments because other implementations might no provide them directly. This information is redundant with the content of `cache_key`. 

The `SQLiteMetadataStore` implementation uses `node_name` and `code_version` explicitly to allow indexing and cache management (e.g., delete all instance of this node). These values could be derived by decoding `cache_key` but that would be a wasteful operation

Reply:
it's more that if people were not using keyword args it would be break due to the change in argument order.

Reply:
(I didn't see any update to callers of this function hence why I asked)